TheJumpCloud/JumpCloud-Serverless
# Cut 4151 role validation check

## Issues
* [CUT-4151](https://jumpcloud.atlassian.net/browse/CUT-4151) - CUT-4151

## What does this solve?
This change solves the second guessing of what permissions and services is needed to build and run the GCP DI app. Additional instructions has been added to enable GCP API services and role permissions for service accounts through the CLI. 
## Is there anything particularly tricky?
n/a
## How should this be tested?
1. On your GCP IAM, remove roles of the service accounts: `PROJECTNUM@cloudbuild.gserviceaccount.com` , and `PROJECTID@appspot.gserviceaccount.com`. 
2. Edit your cloudbuild substitution variables and run `gcloud builds submit` to do a test run, this should fail since your service accounts does not have the proper permissions to build.
3. Follow the instructions on README.md file to build the app
4. Each `gcloud projects add-iam-policy-binding ...` should run without issues if your account have `resourcemanager.projects.setIamPolicy` which project owners have as default
5. When all roles are set to service accounts and api services enabled, build should run successfully.
<img width="609" alt="image" src="https://github.com/user-attachments/assets/f02bd714-64c1-4260-89c5-7cf36816ea4c">



[CUT-4151]: https://jumpcloud.atlassian.net/browse/CUT-4151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
